### DISCUSSION TITLE
How do I send cv output to aws, azure or other cloud services?

### DISCUSSION THREAD

**Discussion by thomedw** [AxisCommunications/acap-computer-vision-sdk-examples#142] [Q&A] [answered]
Hi,

I have browsed through cloud and computer vision examples. How do I send images detected by the cv to the cloud? Thank you.

Best regards,
Thomas

**Comment by Corallo** [comment]
Hello @thomedw 

The container runs an ubuntu system, you can send frames to the cloud just in the same way you would do from any other linux device.
Maybe you can be more specific? Do you want to send a videostream to the cloud? Or frames modified by your application? To something like an S3 bucket, or to something like kinesis?

Here are some options that you can use as an example: [[1]](https://github.com/AxisCommunications/acap-integration-examples-aws/tree/main/images-to-aws-s3) [[2]](https://github.com/AxisCommunications/acap-integration-examples-gcp) [[3]](https://github.com/AxisCommunications/axis-aws-kinesis-vsp)

**Reply by thomedw** [reply]
Hi,

Thank you for your reply. Previously I read about Rule Engine in [the architecture](https://github.com/AxisCommunications/acap-integration-examples-gcp/blob/main/images-to-google-cloud-storage/assets/architecture.png). I think I could send the images to the cloud directly without rule engine. By the way, is there any documentation about how to send images to the rule engine? Thank you.

**Reply by Corallo** [reply, marked as answer]
The rules can be used and set in the camera events tab (in the camera gui). It is explained better [here](https://github.com/AxisCommunications/acap-integration-examples-aws/tree/main/images-to-aws-s3#configure-the-camera). This is however has nothing to do with the containers of the computer vision SDK, it is an API of the [native SDK](https://github.com/AxisCommunications/acap-native-sdk-examples). The events can be triggered for instance when the camera detects movement. Here's the documentation for [axis event](https://axiscommunications.github.io/acap-documentation/docs/api/native-sdk-api.html#event-api)

With the computer vision SDK, you can do something similar by installing an extra python library in the dockerfile of your application, and send events via HTTP by yourself
